### PR TITLE
Fix #12: Better sbt configuration that does not pollute `compile`.

### LINKED ISSUE #12: Incomplete setup documentation
For the shapeless build for 2.12.0-M5 I followed the instructions in the README and saw the following,

```
[error] (coreJS/compile:compileIncremental)
  scala.reflect.internal.MissingRequirementError:
    object org.junit.Test in compiler mirror not found.
```

both when building for the JVM and JS. The fix in my case was to depend on junit-interface everywhere,

```
@@ -60,9 +60,9 @@ lazy val commonSettings = Seq(
 def configureJUnit(crossProject: CrossProject) = {
   crossProject
   .jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin))
-  .jvmSettings(
+  .settings(
     libraryDependencies +=
-      "com.novocode" % "junit-interface" % "0.9" % "test"
+      "com.novocode" % "junit-interface" % "0.9"
   )
 }
```

### PR DESCRIPTION
**Pull request by sjrd**
We do not want to add the compiler plugin in the `compile` configuration. We only want it in the `test` configuration.

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: README.md [hunk 1]
```diff
@@ -17,9 +17,28 @@ As the warning tells you, this will be fixed if you upgrade to JUnit 5 ... excep
...
+    // This is a hack to filter out the dependencies of the plugins
+    if jarPath.contains("plugin")
+  } yield {
+    s"-Xplugin:$jarPath"
+  }
+}
+
+// Add the scala-junit-mixin-plugin to the test configuration when on 2.12
 libraryDependencies ++= {
   if (scalaVersion.value.startsWith("2.12."))
```
Comment (nicolasstucki) [inline README.md]: I would write this condition the other way around to support future versions.

``` scala
val version = scalaVersion.value
if (version.startsWith("2.10.") || version.startsWith("2.11.")) Seq.empty
else Seq("org.scala-js" % "scala-junit-mixin-plugin" % "0.1.0" % "test-plugin" cross CrossVersion.full)
```

**Comment by nicolasstucki** [comment]
That is all.

**Comment by sjrd** [comment]
Updated.

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**